nifi-nar-bundles/nifi-accumulo-bundle/nifi-accumulo-services/src/main/java/org/apache/nifi/accumulo/controllerservices/AccumuloService.java

```diff
@@ -150,61 +204,108 @@ private AuthenticationToken getToken(final AuthenticationType type, final Config
                 }
                 break;
             default:
-                problems.add(new ValidationResult.Builder().valid(false).subject(ACCUMULO_PASSWORD.getName()).explanation("Non supported Authentication type").build());
+                problems.add(new ValidationResult.Builder().valid(false).subject(AUTHENTICATION_TYPE.getName()).explanation("Non supported Authentication type").build());
         }
 
         return problems;
     }
 
     @OnEnabled
     public void onEnabled(final ConfigurationContext context) throws InitializationException, IOException, InterruptedException {
-        if (!context.getProperty(INSTANCE_NAME).isSet() || !context.getProperty(ZOOKEEPER_QUORUM).isSet() || !context.getProperty(ACCUMULO_USER).isSet()){
+        if (!context.getProperty(INSTANCE_NAME).isSet() || !context.getProperty(ZOOKEEPER_QUORUM).isSet()) {
             throw new InitializationException("Instance name and Zookeeper Quorum must be specified");
         }
 
-
-
+        final KerberosCredentialsService kerberosService = context.getProperty(KERBEROS_CREDENTIALS_SERVICE).asControllerService(KerberosCredentialsService.class);
         final String instanceName = context.getProperty(INSTANCE_NAME).evaluateAttributeExpressions().getValue();
         final String zookeepers = context.getProperty(ZOOKEEPER_QUORUM).evaluateAttributeExpressions().getValue();
-        final String accumuloUser = context.getProperty(ACCUMULO_USER).evaluateAttributeExpressions().getValue();
+        final AuthenticationType authType = AuthenticationType.valueOf( context.getProperty(AUTHENTICATION_TYPE).getValue());
+
+        final Properties clientConf = new Properties();
+        clientConf.setProperty("instance.zookeepers", zookeepers);
+        clientConf.setProperty("instance.name", instanceName);
 
-        final AuthenticationType type = AuthenticationType.valueOf( context.getProperty(AUTHENTICATION_TYPE).getValue() );
+        switch(authType){
+            case PASSWORD:
+                final String accumuloUser = context.getProperty(ACCUMULO_USER).evaluateAttributeExpressions().getValue();
 
+                final AuthenticationToken token = new PasswordToken(context.getProperty(ACCUMULO_PASSWORD).getValue());
 
+                this.client = Accumulo.newClient().from(clientConf).as(accumuloUser, token).build();
+                break;
+            case KERBEROS:
+                final String principal = kerberosService == null ? context.getProperty(KERBEROS_PRINCIPAL).getValue() : kerberosService.getPrincipal();
 
-        final AuthenticationToken token = getToken(type,context);
+                if (kerberosService != null) {
+                    this.kerberosUser = new KerberosKeytabUser(principal, kerberosService.getKeytab());
+                } else {
+                    this.kerberosUser = new KerberosPasswordUser(principal, context.getProperty(KERBEROS_PASSWORD).getValue());
+                }
```

Comment:
Optional, although I think it's much cleaner this way.

```suggestion
                if (kerberosService == null) {
                    principal = context.getProperty(KERBEROS_PRINCIPAL).getValue();
                    this.kerberosUser = new KerberosPasswordUser(principal, context.getProperty(KERBEROS_PASSWORD).getValue());
                } else {
                    principal = kerberosService.getPrincipal();
                    this.kerberosUser = new KerberosKeytabUser(principal, kerberosService.getKeytab());
                }
```